lecture_4/homework/app/controllers/mercenaries_controller.rb

```diff
@@ -3,30 +3,20 @@
 # This controller is written badly on purpose. Please refactor this
 class MercenariesController < ApplicationController
   def index
-    render json: Mercenary.where('available_from < ?', Time.now).all
+    mercenaries = params[:alive] ? Mercenary.alive : Mercenary.all
+    mercenaries = MercenariesQuery.unavailable(relation: mercenaries) if params.has_key?(:unavailable)
+    mercenaries = MercenariesQuery.available(relation: mercenaries) if params.has_key?(:available)
+    return render json: { "error": "There are no mercenaries available right now" } if mercenaries.blank?
+    render json: mercenaries
   end
 
   def show
     render json: mercenary, include: [:warrior]
   end
 
-  def employ_best
-    mercenary = Mercenary.where('available_from < ?', Time.now).order(price: :asc).first # TODO: what about experience?
-    clan = find_clan
-    building = find_building
-    warrior_class = clan.warriors.select('type, count(type) as warriors_count').group(:type).order('warriors_count ASC').first.class
-    warrior = warrior_class.create!(name: mercenary.name, clan: clan, building: building, preferred_weapon_kind: mercenary.preferred_weapon_kind, mercenary: mercenary)
-    create_good_weapon(mercenary)
-    render json: warrior, include: [:mercenary], status: 201
-  end
+  def create
+    warrior = WarriorCreator.new(params: mercenary_params).call
 
```

Comment:
akcje employ_best i employ wykonują dokładnie to samo (zatrudniają najemnika) z tą różnicą, że wykonują się w innych miejscach aplikacji. żeby trzymać się restowego standardu stworzyłem akcję create i przekierowałem je obie tutaj. jeżeli akcja create nie otrzyma id, to wyszuka najbardziej doświadczonego najemnika